packages/ag-charts-community/src/chart/tooltip/tooltip.ts

```diff
@@ -436,13 +436,19 @@ export class Tooltip extends BaseProperties {
     private toggle(visible: boolean) {
         if (!this.element?.isConnected) return;
 
+        // Avoid touching the DOM if invisible and visibility status hasn't changed.
+        if (!this._visible && !visible) return;
```

Comment:
Did this check improve perf? It feels like it'll be tricky to get this one right because it also affects timeouts

Reply:
Yes, this code path was clearly showing up as a hot-spot due to (usually redundant) DOM updates before this change.